### DISCUSSION TITLE
Testing SO: activity not showing up in alerts until almost an hour later, and seems incomplete.

### DISCUSSION THREAD

**Discussion by bilkoh** [Security-Onion-Solutions/securityonion#2104] [Unsupported Versions] [answered]
I'm trying to run test activity that'll show up as alerts. Stuff like nmap scans, opening up backdoors, metasploitable activity. The activity seems incomplete. I just want verification that my problem isn't of a traffic capture nature.

After nearly an hour, some activity finally showed up:
![image](https://user-images.githubusercontent.com/43228593/100283964-3dac6080-2f23-11eb-9cd8-e97f045052ca.png)

Running Security Onion 2.3.2. Everything is running on vlan10. The machine has 12gb ram / 4 sockets / 320gb storage. The environment is virtualized, and my span port is setup through Open vSwitch.

I can see my span interface picking up packets in from the vlan tcpdump.

`so-test` seems normal:
~~~
Statistics for network device: bond0
        Successful packets:        55304
        Failed packets:            444
        Truncated packets:         0
        Retried packets (ENOBUFS): 0
        Retried packets (EAGAIN):  0
~~~

I ran `curl https://testmyids.com/` on a VM as well.

Is this normal?

**Comment by weslambert** [comment]
The only way you can really know if there is an issue with the sniffing or the rule implementation itself, is to test known "bad" against a rule so that it fires on exactly that traffic you are throwing at the interface and for which you expect it to trigger.  You'll want to first confirm that you can manually trigger a specific rule or multiple rules, then send that same traffic across the interface (or confirm observation of the similarities that should trigger the alert)  Otherwise, you are just guessing, and could potentially go down many different rabbit holes trying to find the answer.  

With regard to the curl command, there are also several variables that could impact your results, and the execution and recorded response (from what SO was able to ascertain) should sometimes be taken with a grain of salt.  One option would be to use something like Scapy and create a rule specific to the traffic contained within a crafted pcap.  

We have an example here: https://docs.securityonion.net/en/16.04/alerts.html?highlight=scapy#testing-to-make-sure-the-ids-is-working that has not yet been adapted for Security Onion 2, but you should get the idea.  Local rules in Security Onion 2 can be added by following the guidance here: https://docs.securityonion.net/en/latest/local-rules.html?highlight=lcaol%20rules  Let me know if you have any questions.  

Otherwise, `so-test` _should_ generate some alerts for you to view in the interface.

Thanks, Wes

**Comment by bilkoh** [comment]
Thank you @weslambert.
So I ran the scapy pcap, so-test, curl and tmNIDS bash script.

All of it showed up in wireshark, like so:
![image](https://user-images.githubusercontent.com/43228593/100308122-69980800-2f5c-11eb-9cc3-08b863cdfcd2.png)

And an hour later I'm not seeing any of it show up within the alerts page.

Any ideas of where to go next?

Appreciate the help.

**Reply by bilkoh** [reply]
Ok I have an update. After 90 minutes they showed up. Is this delay typical? Is it a matter of not enough computing resources? Or some misconfiguration?

The snapy alert did not show up. I tried to follow `https://docs.securityonion.net/en/2.3/local-rules.html` for adding local rules but `/opt/so/saltstack/local/salt/idstools/localrules/local.rules` did not exist for me. So I added the rule to `/opt/so/saltstack/local/salt/idstools/local.rules` instead. Ran `salt $SENSORNAME_$ROLE state.apply suricata` and then used `so-import-pcap` to import it. And like I said before, Wireshark detected this. I could even it load it in Hunt. So I think some how I'm not able to add the local rule correctly. Can anyone tell me what I did wrong?

**Reply by greatapoc** [reply]
2.3.10 fixed the issues I was experiencing with relative time queries not working correctly. It was due to a bug with timezones.

**Reply by bilkoh** [reply]
Ah. That makes more sense. Thank you.

I had reset the link to /etc/localtime. Is there anything else you did to set the correct timezone @greatapoc.

**Reply by greatapoc** [reply]
Nope it was all correct from when it was specified during the CentOS installation.

**Comment by bilkoh** [comment]
Thanks everyone. I think I've narrowed this issue to a matter of timezone/time.

Since I'm running this as a virtual machine within proxmox, the hypervisor perfers it's VMs to sync itself with the hypervisor's clock. That's my thinking. I tried various fixes after changing my timezone:
- Toggled chrony off and on
- toggled my hardware clock to local tz with `timedatectl set-local-rtc 1` and back

When I reboot, the timedatectl status looks accurate:
~~~
[bilk0h@onion ~]$ timedatectl status
      Local time: Thu 2020-11-26 03:00:53 PST
  Universal time: Thu 2020-11-26 11:00:53 UTC
        RTC time: Fri 2020-11-27 03:00:52
       Time zone: America/Los_Angeles (PST, -0800)
     NTP enabled: yes
NTP synchronized: yes
 RTC in local TZ: no
      DST active: no
~~~

But about 2-3 minutes after boot up, it reverts back to UTC for some reason:
~~~
[bilk0h@onion ~]$ timedatectl status
      Local time: Thu 2020-11-26 11:02:06 UTC
  Universal time: Thu 2020-11-26 11:02:06 UTC
        RTC time: Fri 2020-11-27 03:02:05
       Time zone: Etc/UTC (UTC, +0000)
     NTP enabled: yes
NTP synchronized: yes
 RTC in local TZ: no
~~~

Anyone know what mechanism is reverting the time setting back to UTC?

**Comment by dougburks** [comment, marked as answer]
Security Onion intentionally sets the OS timezone to UTC:
https://docs.securityonion.net/en/2.3/timezones.html

Security Onion 2.3.2 had a timezone bug which has been resolved in 2.3.10, so please make sure you're running 2.3.10. From https://docs.securityonion.net/en/2.3/release-notes.html#changes:
> Updated Go to newer version to fix a timezone, daylight savings time (DST) issue that resulted in Alerts and Hunt interfaces not consistently showing results.
